### PR TITLE
feat: token detection on 7 more networks

### PR DESCRIPTION
**Pull request by bergeron**
## **Description**

Adds support for erc20 token detection on 7 more networks.

Patching this from core because it's small: https://github.com/MetaMask/core/pull/4184

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24121?quickstart=1)

## **Related issues**

https://consensyssoftware.atlassian.net/browse/MMASSETS-207

## **Manual testing steps**

1. Visit one of the new chains (cronos, celo, gnosis, fantom, polygon zk, moonbeam, moonriver) on a wallet that has tokens on the network
2.  They should be autodetected

### **Before**

None of the chains support token lists or detection:

Screenshot 2024-04-22 at 9 50 03 PM

### **After**

All 7 chains support detection:

https://github.com/MetaMask/metamask-extension/assets/3500406/16c8e763-ccc0-4070-8ac3-28a7dfa88e1f

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .yarn/patches/@metamask-assets-controllers-npm-26.0.0-17c0e9432c.patch [hunk 1]
```diff
@@ -1,24 +1,56 @@
-diff --git a/dist/TokenDetectionController.js b/dist/TokenDetectionController.js
-index 9aa09140d47424217eac118aebca9031e5d2a236..8100c432e7e01dbefcb5f53db3c58e51f120a51d 100644
---- a/dist/TokenDetectionController.js
-+++ b/dist/TokenDetectionController.js
-@@ -230,8 +230,7 @@ _TokenDetectionController_intervalId = new WeakMap(), _TokenDetectionController_
-         const isDetectionChangedFromPreferences = __classPrivateFieldGet(this, _TokenDetectionController_isDetectionEnabledFromPreferences, "f") !== useTokenDetection;
-         __classPrivateFieldSet(this, _TokenDetectionController_selectedAddress, newSelectedAddress, "f");
-         __classPrivateFieldSet(this, _TokenDetectionController_isDetectionEnabledFromPreferences, useTokenDetection, "f");
--        if (useTokenDetection &&
```
Comment (bergeron) [inline .yarn/patches/@metamask-assets-controllers-npm-26.0.0-17c0e9432c.patch:9]: didnt mean to undo this.  need cleaner way to patch on top of patch

File: package.json [hunk 2]
```diff
@@ -271,7 +271,7 @@
     "@metamask/address-book-controller": "^3.1.7",
     "@metamask/announcement-controller": "^6.1.0",
     "@metamask/approval-controller": "^6.0.0",
-    "@metamask/assets-controllers": "patch:@metamask/assets-controllers@patch%3A@metamask/assets-controllers@npm%253A26.0.0%23~/.yarn/patches/@metamask-assets-controllers-npm-26.0.0-17c0e9432c.patch%3A%3Aversion=26.0.0&hash=cf1d54#~/.yarn/patches/@metamask-assets-controllers-patch-0f46262fea.patch",
+    "@metamask/assets-controllers": "patch:@metamask/assets-controllers@npm%3A26.0.0#~/.yarn/patches/@metamask-assets-controllers-npm-26.0.0-17c0e9432c.patch",
```
Comment (bergeron) [inline package.json:274]: I don't know if this is right.  I did `yarn patch @metamask/assets-controllers` like usual.  But idk if you need to do something different to patch on top of a patch.

Comment (sahar-fehri) [reply package.json:274]: That was automatically generated with `yarn patch @metamask/assets-controllers --update`
Can hop on a call to check this; i have another ongoing PR that patches same controller

Comment (bergeron) [reply package.json]: resolved offline

**Comment by sahar-fehri** [comment]
I came across this behavior where if i change the network the import does not work anymore... and you would have to refresh the page again to get tokens/icons ... this might not be related to this PR 

https://github.com/MetaMask/metamask-extension/assets/10994169/3c303f9e-5fc2-4e4a-ab5a-d2c818080c25

**Comment by sahar-fehri** [comment]
Alright confirming that the behavior is not related to this PR. Can reproduce on another branch

**Review by sahar-fehri — APPROVED** [review]

**Comment by bergeron** [comment]
> I came across this behavior where if i change the network the import does not work anymore... and you would have to refresh the page again to get tokens/icons ... this might not be related to this PR
> 
>  Screen.Recording.2024-04-24.at.14.41.44.mov

fixed here: https://github.com/MetaMask/metamask-extension/pull/24212

**Review by salimtb — APPROVED** [review]

**Pull request merged**